src/app/AlwaysOn.Shared/TelemetryExtensions/AppInsightsRequestHandler.cs

```diff
@@ -0,0 +1,73 @@
+            var title = request.Properties.ContainsKey("Operation") ? request.Properties["Operation"].ToString() : $"{request.Method} {request.RequestUri.OriginalString}";
+            var dbClientEndpoint = request.Properties.ContainsKey("DbClientEndpoint") ? request.Properties["DbClientEndpoint"].ToString() : null;
+
+            using var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(title);
+
+            //
+            // Comparison with the original telemetry object:
+            //
+            //var telemetry = new DependencyTelemetry()
+            //{
+            //    OK - Type = AppInsightsDependencyType,
+            //    OK - Data = $"ObjectId={objectId}, Partitionkey={partitionKey}",
+            //    OK - Name = $"Delete {typeof(T).Name}",
+            //    OK - Timestamp = startTime,
+            //    OK - Duration = diagnostics != null ? diagnostics.GetClientElapsedTime() : overallDuration,
+            //    OK - Target = diagnostics != null ? diagnostics.GetContactedRegions().FirstOrDefault().uri?.Host : _dbClient.Endpoint.Host,
+            //    REMOVED - Success = success
+            //};
+            //
+            //if (response != null)
+            //    OK - telemetry.Metrics.Add("CosmosDbRequestUnits", response.RequestCharge);
+
+            var response = await base.SendAsync(request, cancellationToken);
+
+            var success = response.IsSuccessStatusCode;
+            // Application-specific special handling in order to have cleaner Application Insights error reporting. Client will still get proper error response.
+            // There are cases which we don't consider database errors and would create noise in monitoring:
+            //  - Considering NotFound as successful call to Cosmos DB, looking for a non-existent object.
+            //  - The same with Conflict, which is handled the application code (ignoring the request, because it already exists).
+            if (response.StatusCode == System.Net.HttpStatusCode.NotFound || response.StatusCode == System.Net.HttpStatusCode.Conflict)
+                success = true;
+
+            dependency.Telemetry.Type = "Azure DocumentDB";
+            dependency.Telemetry.Data = request.RequestUri.OriginalString; // Will be shown as "Command" in Application Insights
+            dependency.Telemetry.Target = response.Diagnostics != null ? response.Diagnostics.GetContactedRegions().FirstOrDefault().uri?.Host : dbClientEndpoint;
+            dependency.Telemetry.ResultCode = ((int)response.StatusCode).ToString();
+            dependency.Telemetry.Success = success;
+
+            dependency.Telemetry.Metrics.Add("CosmosDbRequestUnits", response.Headers.RequestCharge);
+            dependency.Telemetry.Metrics.Add("ClientElapsedTime", response.Diagnostics != null ? response.Diagnostics.GetClientElapsedTime().TotalMilliseconds : -1);
+
+            if (request.Headers.TryGetValue("x-ms-documentdb-partitionkey", out string partitionKey))
```

Comment:
I guess you checked if this isnt part of the Diagnostics object?!

Reply:
Yep and it isn't.